### PR TITLE
Preserve EMA updates when resuming training

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
## What has changed and why?

When training resumes from an interrupted checkpoint, `ModelEMA.updates` is currently reset to zero because it is stored as a plain Python integer and therefore is not included in the checkpoint state dict.

Resetting the counter restarts the EMA decay warmup. On the first update after resuming, the decay is close to zero, so the accumulated EMA weights are almost completely overwritten by the raw model weights. Validation uses the EMA model, which causes an artificial drop in validation metrics immediately after resume.

This PR registers `updates` as a scalar tensor buffer. The counter is then saved and restored together with the EMA state, preserving the decay schedule and accumulated EMA weights across interrupted runs.

Previously exported `.pt` files can still be loaded without raising an error. Exported files contain the inference task model rather than the training-only `ModelEMA` wrapper, so the newly registered buffer does not change their state dict.

## How has it been tested?

- A smoke test with a real `train_instance_segmentation` -> save `last.ckpt` -> `resume_interrupted=True` cycle (`_ltdetrv2-seg-s-notpretrained`, tiny synthetic COCO data, `grad_accum=1`, `ema_warmup_steps=2` so the EMA saturates within the short run — the regime where the bug is visible). 

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/object_detection_components/ema.py [hunk 1]
```diff
@@ -47,7 +47,13 @@ def __init__(
...
         self.decay = decay
         self.warmups = warmups
-        self.updates = 0  # number of EMA updates
+        # Track the number of EMA updates as a buffer so it is saved in the model
+        # state_dict and restored on resume. If it were a plain Python int it would
+        # reset to 0 when resuming from a checkpoint, restarting the decay warmup ramp
+        # (decay_fn(step=1) ~= 0) and overwriting the accumulated EMA weights with the
+        # raw model weights on the first post-resume update. Since validation uses the
+        # EMA weights, that produces a spurious drop in val metrics after resuming.
+        self.register_buffer("updates", torch.zeros((), dtype=torch.long))
```
[the remark this answers is not in the document]
Comment (yutong-xiang-97) [reply src/lightly_train/_task_models/object_detection_components/ema.py:59]: This only happens when you try to resume from interrupted using `.ckpt` with a different released version.

**Review by liopeer — APPROVED** [review]
Amazing find!
